validator_client/src/lib.rs

```diff
@@ -331,15 +331,18 @@ impl<T: EthSpec> ProductionValidatorClient<T> {
             .collect::<Result<Vec<BeaconNodeHttpClient>, String>>()?;
 
         let num_nodes = beacon_nodes.len();
+
         let candidates = beacon_nodes
             .into_iter()
-            .map(CandidateBeaconNode::new)
+            .zip(0..num_nodes)
+            .map(|(node, id)| CandidateBeaconNode::new(node, id))
```

Comment:
```suggestion
            .enumerate()
            .map(|(id, node)| CandidateBeaconNode::new(node, id))
```

This might be more idiomatic?